core/src/com/unciv/logic/UncivScript.kt

```diff
@@ -0,0 +1,582 @@
+            ConfirmPopup(
+                popupScreen,
+                "{Script output}:\n\n${responseString}\n{in}\n${scriptPath}",
+                "Copy contents to clipboard",
+                true
+            ) {
+                Gdx.app.clipboard.contents = responseString
+                errorReportCopiedToast(popupScreen)
+            }.open()
+        }
+    }
+
+    private fun filterResponses(responses: List<ScriptDiagnostic>?,
+                                minimumReportThreshold: ScriptDiagnostic.Severity = ScriptDiagnostic.Severity.WARNING
+    ): List<ScriptDiagnostic>? {
+        if (responses == null) return null
+
+        // stack traces contain lines that are irrelevant to modders so only take lines that contain
+        // a mention of .unciv.kts unless they're a non-error, non-exception line
+        val newResponses = arrayListOf<ScriptDiagnostic>()
+        for (response in responses) {
+            if (response.severity < ScriptDiagnostic.Severity.ERROR) {
+                newResponses.add(response) // add non-error and non-exception responses as-is
+                continue
+            }
+
+            var newMessage = ""
+            var lineCount = 0
+            val splitMessage = response.message.split("\n\t")
+            for (segment in splitMessage) {
+                lineCount += 1
+                if (newMessage == "") {
+                    newMessage += segment // the actual exception text
+                    continue
+                }
+                if (segment.contains(".unciv.kts")) {
+                    newMessage +="\n\t$segment"
+                }
+            }
+            if (lineCount > 1) {
+                /**
+                 * The last error line will reference a line that doesn't exist in the modder's original
+                 * source code for an intentional reason (that line contains the stack trace mechanism)
+                 * so truncate the messsage so modders don't get confused as to why there's an exception
+                 * on line 102 when they only wrote 100 lines of code so get rid of it to prevent
+                 * endless "Is this a bug?" questions.
+                 */
+                newMessage = newMessage.split("\n\t").dropLast(1).joinToString("\n\t")
+            }
+            newResponses.add(
+                ScriptDiagnostic(response.code, newMessage, response.severity,
+                response.sourcePath, response.location, response.exception)
+            )
+        }
+
+        return newResponses
+            .filterNot { it.message.startsWith("Using new faster version of JAR FS") } // generic compiler warning
+            .filter { it.severity >= minimumReportThreshold }
+            .sortedByDescending { it.severity }
+    }
+}
+
+/**
+ * Inherits [CompiledScriptJarsCache] to modify caching so we can save in the jar manifest then check
+ * Unciv version for any potential incompatibilities.
+ */
+@Suppress("UNCHECKED_CAST")
```

Comment:
Below this point, there's tons of code that is subject to removal. This is an experimental feature within the Kotlin engine with very little documentation so I had to do a lot of testing and debugging to figure out how things worked.